bin/resty

```diff
@@ -360,13 +360,23 @@ if ($use_valgrind) {
 
 my $child_pid;
 
-sub sigint {
-    $SIG{INT} = \&sigint;
+sub forward_signal {
+    my $signame = shift;
+
     if ($child_pid) {
-        kill INT => $child_pid;
+        unless(kill $signame => $child_pid) {
+            die "child process is dead, can't forward signal to it";
```

Comment:
I think you should use the system error message instead of guessing the reason here ourselves. The `kill` function sets the error string to the special variable `$!`.

Also, it's more idiomatic to say

```perl
kill $signame => $child_pid
    or die "failed to send $signame to process $child_pid: $!";
```